arch/cortex-m4/src/mpu.rs

```diff
@@ -163,8 +163,18 @@ impl kernel::mpu::MPU for MPU {
         } else {
             // Memory base not aligned to memory size
 
-            // Which subregion size would align with the base address?
-            let subregion_size = start % len;
+            // Which (power-of-two) subregion size would align with the base
+            // address?
+            //
+            // We find this by taking smallest binary substring of the base
+            // address with exactly one bit:
+            //
+            //      1 << (start.trailing_zeros())
+            //
+            // The zero case (where there are no 1 bits) is taken care of with
+            // the `wrapping_shl` operation, which masks bits that would
+            // overflow the integer width.
+            let subregion_size = (1 as usize).wrapping_shl(start.trailing_zeros());
```

Comment:
Mhmm, what I don't love about this is that the result of this operation is a subregion size of 1, which is never valid anyway. It feels weird to handle a case that (1) can never happen b/c it's guarded by an if, and (2) returns an invalid result anyway.

I suppose I don't know what the `<<` operator is defined to do in rust, I would expect that `1 << 32` -> 0, giving an equally invalid subregion size? I think the wrapping_shl is just noise here